apps/finance/contracts/Finance.sol

```diff
@@ -507,6 +485,37 @@ contract Finance is EtherTokenConstant, IsContract, AragonApp {
 
     // internal fns
 
+    function _deposit(address _token, uint256 _amount, string _reference, address _sender, bool _isExternalDeposit) internal {
+        require(_amount > 0);
+        _recordIncomingTransaction(
```

Comment:
Maybe we should do the record at the end so if something reverts we don't waste the gas spent in storage here

Reply:
It might not be safe to do so, to avoid reentrancy issues.

Reply:
I can't see how we can avoid reentrancy in this case, if the token contract is malicious this function can re-enter regardless of where we record the transaction